### PR TITLE
Horizon API protocol doc

### PR DESCRIPTION
**Pull request by bartekn**
Created this doc to find out and start tracking which SDK version fully supports given Horizon API version. This should help us update SDKs to support all the endpoints.

[Human readable version](https://github.com/stellar/go/blob/horizon-protocol-doc/protocols/horizon/README.md)

### CODE REVIEW

**Comment by robertDurst** [comment]
I second @tomerweller 's comment, a bit hard to parse through with +/- 

Would Y/N work?

**Comment by bartekn** [comment]
Done.
